client/src/views/NewInterview.js

```diff
@@ -118,15 +122,42 @@ const InterviewPage = () => {
     }
   };
 
-  const warnIfTabNotActive = () => {
-    if (document.hidden) {
-      toast("Warning: Tab switching is not allowed!", {
-        icon: "⚠️",
-      });
-      console.log("Warning: Tab switching is not allowed!");
-    }
+  const onCloseCheatingModal = () => {
+    setIsCheatingModalOpen(false);
+    exitFullScreen();
+    navigate("/thank-you");
   };
 
+  useEffect(() => {
+    const handleFocusChange = () => {
+      if (!document.hasFocus()) {
+        setTabSwitchCount((prev) => prev + 1);
+
+        if (tabSwitchCount > 2) {
+          setIsCheatingModalOpen(true);
+          setTimeout(() => {
+            onCloseCheatingModal();
+          }, 8000);
+        }
+
+        toast(
+          `Warning: Tab switching or window change detected! ${tabSwitchCount}`,
+          {
+            icon: "⚠️",
+          }
+        );
+      }
+    };
+
+    window.addEventListener("focus", handleFocusChange);
+    window.addEventListener("blur", handleFocusChange);
```

Comment:
What is the blur event listener for?

Reply:
Actually, I need to remove the focus Event Listener. When the user switches the tab, it will be detected as blurred, and we can give an alert.